fhirzeug/generators/python_pydantic/templates/resource_header.py

```diff
@@ -95,9 +95,66 @@ def dict(self, *args, **kwargs):
         return _without_empty_items(serialized) or {}
 
     @pydantic.root_validator(pre=True)
-    def strip_empty_items(cls, valuse):
+    def strip_empty_items(cls, values: typing.Dict) -> typing.Dict:
         """This strips all empty elements according to the fhir spec."""
-        return _without_empty_items(valuse) or {}
+        return _without_empty_items(values) or {}
+
+    @pydantic.root_validator()
+    def dynamic_post_root_validator(cls, values: typing.Dict) -> typing.Dict:
+        """Validate data.
+
+        The behavior of this validator can be changed after definition of the BaseModel
+        by using method `_add_post_root_validator`.
+        """
+        for validator in cls._dynamic_validators():
+            values = validator(values)
+        return values
+
+    @classmethod
+    def _add_post_root_validator(
+        cls, validator: typing.Callable[[typing.Dict], typing.Dict]
+    ) -> None:
+        """Add a post root validator to the FHIR object.
+
+        The order of the dynamic validators has not been a priority in this
+        implementation. All dynamic validators must be considered to be independent
+        from each other.
+        TODO : add more flexibility to order validators.
+
+        Internally, each FHIR class stores a list of dynamic validators. Then,
+        `dynamic_post_root_validator` iterates over validators one by one using the
+        __mro__ resolution order.
+
+        Warning: there is currently no way of removing a validator. Since this
+        method is more or less doing monkeypatching, it is preferred to use it
+        carefully.
+        """
+        dynamic_validators_field = cls._get_dynamic_validators_field_name()
+        dynamic_validators = getattr(cls, dynamic_validators_field, [])
+        dynamic_validators.append(validator)
+        setattr(cls, dynamic_validators_field, dynamic_validators)
+
+    @classmethod
+    def _dynamic_validators(
+        cls,
+    ) -> typing.Generator[typing.Callable[[typing.Dict], typing.Dict], None, None]:
+        """Return a generator iterating over dynamic validators."""
+        for subclass in cls.__mro__:
+            if issubclass(subclass, FHIRAbstractBase):
```

Comment:
As a micrroptimization we could add an `else` clause:

```
else:
    return
```

Returning there is safe, because we're already in the parent classes of `FHIRAbstractBase`.

Reply:
Ahah, indeed